examples/erc1155/lib.rs

```diff
@@ -0,0 +1,742 @@
+            );
+
+            assert_eq!(
+                token_ids.len(),
+                values.len(),
+                "The number of tokens being transferred ({:?}) does not match the number of transfer amounts ({:?}).",
+                token_ids.len(), values.len()
+            );
+
+            token_ids.iter().zip(values.iter()).for_each(|(&id, &v)| {
+                self.perform_transfer(from, to, id, v, data.clone());
+            })
+        }
+
+        #[ink(message)]
+        fn balance_of(&self, owner: AccountId, token_id: TokenId) -> Balance {
+            *self.balances.get(&(owner, token_id)).unwrap_or(&0)
+        }
+
+        #[ink(message)]
+        fn balance_of_batch(
+            &self,
+            owners: Vec<AccountId>,
+            token_ids: Vec<TokenId>,
+        ) -> Vec<Balance> {
+            let mut output = Vec::new();
+            for o in &owners {
+                for t in &token_ids {
+                    let amt = self.balance_of(*o, *t);
+                    output.push(amt);
+                }
+            }
+            output
+        }
+
+        #[ink(message)]
+        fn set_approval_for_all(&mut self, operator: AccountId, approved: bool) {
+            let caller = self.env().caller();
+
+            assert!(
+                operator != caller,
+                "An account does not need to approve themselves to transfer tokens."
+            );
+
+            let approval = Approval {
+                owner: caller,
+                operator,
+            };
+
+            if approved {
+                self.approvals.insert(approval, ());
+            } else {
+                self.approvals.remove(&approval);
+            }
+
+            self.env().emit_event(ApprovalForAll {
+                owner: approval.owner,
+                operator,
+                approved,
+            });
+        }
+
+        #[ink(message)]
+        fn is_approved_for_all(&self, owner: AccountId, operator: AccountId) -> bool {
+            self.approvals.get(&Approval { owner, operator }).is_some()
+        }
+    }
+
+    impl super::Erc1155TokenReceiver for Contract {
+        #[ink(message, selector = "0xF23A6E61")]
+        fn on_received(
+            &mut self,
+            _operator: AccountId,
+            _from: AccountId,
+            _token_id: TokenId,
+            _value: Balance,
+            _data: Vec<u8>,
+        ) -> Vec<u8> {
+            unimplemented!("This smart contract does not accept token transfer.")
+        }
+
+        #[ink(message, selector = "0xBC197C81")]
+        fn on_batch_received(
+            &mut self,
+            _operator: AccountId,
+            _from: AccountId,
+            _token_ids: Vec<TokenId>,
+            _values: Vec<Balance>,
+            _data: Vec<u8>,
+        ) {
+            unimplemented!("This smart contract does not accept batch token transfers.")
+        }
+    }
+
+    /// Unit tests.
+    #[cfg(not(feature = "ink-experimental-engine"))]
```

Comment:
There are some differences in the APIs.

I've added some conditional compilation flags so that testing works in the experimental environment. If you think this approach is okay we should update other contracts which have this restriction (e.g the `Erc20` example tests)

Reply:
Generally it is okay to just support the new engine since it will replace the old one.

Reply:
I think it's best to leave both for now. If people are meant to use the examples as a reference for how to implement certain things in their own contracts we should show them both the old and new way - at least until the experimental engine is not experimental anymore.

There's also a few other example contracts which support both, such as `contract-terminate`. If we are going to only support a single engine we should remove support from all the example contracts, which would best be done in a follow up PR.